packages/material-ui/src/Theme/MaterialUIContext.tsx

```diff
@@ -1,92 +1,130 @@
+    MaterialUIContext.FormGroup = FormGroup.default;
+  });
+  import('@material-ui/core/FormHelperText').then(FormHelperText => {
+    MaterialUIContext.FormHelperText = FormHelperText.default;
+  });
+  import('@material-ui/core/FormLabel').then(FormLabel => {
+    MaterialUIContext.FormLabel = FormLabel.default;
+  });
+  import('@material-ui/core/Grid').then(Grid => {
+    MaterialUIContext.Grid = Grid.default;
+  });
+  import('@material-ui/core/IconButton').then(IconButton => {
+    MaterialUIContext.IconButton = IconButton.default;
+  });
+  import('@material-ui/core/OutlinedInput').then(OutlinedInput => {
+    MaterialUIContext.Input = OutlinedInput.default;
+  });
+  import('@material-ui/core/InputLabel').then(InputLabel => {
+    MaterialUIContext.InputLabel = InputLabel.default;
+  });
+  import('@material-ui/core/List').then(List => {
+    MaterialUIContext.List = List.default;
+  });
+  import('@material-ui/core/ListItem').then(ListItem => {
+    MaterialUIContext.ListItem = ListItem.default as ComponentType;
+  });
+  import('@material-ui/core/ListItemIcon').then(ListItemIcon => {
+    MaterialUIContext.ListItemIcon = ListItemIcon.default;
+  });
+  import('@material-ui/core/ListItemText').then(ListItemText => {
+    MaterialUIContext.ListItemText = ListItemText.default;
+  });
+  import('@material-ui/core/MenuItem').then(MenuItem => {
+    MaterialUIContext.MenuItem = MenuItem.default as ComponentType;
+  });
+  import('@material-ui/core/Paper').then(Paper => {
+    MaterialUIContext.Paper = Paper.default;
+  });
+  import('@material-ui/core/Radio').then(Radio => {
+    MaterialUIContext.Radio = Radio.default;
+  });
+  import('@material-ui/core/RadioGroup').then(RadioGroup => {
+    MaterialUIContext.RadioGroup = RadioGroup.default;
+  });
+  import('@material-ui/icons/Remove').then(Remove => {
+    MaterialUIContext.RemoveIcon = Remove.default;
+  });
+  import('@material-ui/core/Slider').then(Slider => {
+    MaterialUIContext.Slider = Slider.default;
+  });
+  import('@material-ui/core/SvgIcon').then(SvgIcon => {
+    MaterialUIContext.SvgIcon = SvgIcon.default;
+  });
+  import('@material-ui/core/TextField').then(TextField => {
+    MaterialUIContext.TextField = TextField.default;
+  });
+  import('@material-ui/core/Typography').then(Typography => {
+    MaterialUIContext.Typography = Typography.default;
+  });
 } catch (err) {
-  // purposely a no-op. If it is not here, don't make noise like we used to
+  // no-op
 }
 
-export let MaterialUIContext: MaterialUIContextProps;
-export let DefaultChildren = () => <div>@material-ui not available</div>;
+export const DefaultChildren = () => {
+  if (
+    MaterialUIContext !== defaultMaterialUIContext &&
+    MaterialUIContext.Box &&
+    MaterialUIContext.Button &&
+    MaterialUIContext.AddIcon
+  ) {
+    const { Box, Button } = MaterialUIContext;
+    const BoxComp = Box as ComponentType<BoxProps>;
+    const ButtonComp = Button as ComponentType<ButtonProps>;
```

Comment:
Won't these same casts also need to be done deeper into the rest of the code?

Reply:
It certainly is if you want intellisense and static typing in the IDE at time of dev. The `any` props passed in allows for all. What I did with the `as` callout really isn't necessary...it's just a habit (and a good one, I admit).